### PR TITLE
Migrate build/deploy to 1es

### PR DESCRIPTION
**Pull request by rkodev**
Migrate build/deploy to 1es

Resolves https://github.com/microsoft/kiota-cli-commons/issues/160

### CODE REVIEW

File: .azure-pipelines/ci-build.yml [hunk 1]
```diff
@@ -21,248 +18,252 @@ variables:
...
+            displayName: 'Use .NET 8'
+            inputs:
+              version: 8.x
 
-      - task: PoliCheck@1
-        displayName: 'Run PoliCheck "/src"'
-        inputs:
-          inputType: CmdLine
-          cmdLineArgs: '/F:$(Build.SourcesDirectory)/src /T:9 /Sev:"1|2" /PE:2 /O:poli_result_src.xml'
+          - task: PoliCheck@1
```
Comment (andrueastman) [inline .azure-pipelines/ci-build.yml:48]: The steps like PoliCheck/Binskim/CredScan are auto-inject by 1es and should be removed from the pipeline definition. Checkout the PR for a similar change https://github.com/microsoft/kiota-abstractions-dotnet/pull/204/files

File: .azure-pipelines/ci-build.yml [hunk 2]
```diff
@@ -21,248 +18,252 @@ variables:
...
+                    artifact: Nugets
+                    source: current
+                - task: PowerShell@2
+                  displayName: 'Extract release information to pipeline'
+                  inputs:
+                    targetType: 'filePath'
+                    filePath: $(Pipeline.Workspace)\Nugets\scripts\GetNugetPackageVersion.ps1
+                    pwsh: true
+                    arguments: '-packageDirPath "$(Pipeline.Workspace)/Nugets/"'
+                - task: NuGetCommand@2
```
Comment (andrueastman) [inline .azure-pipelines/ci-build.yml:263]: This has to be as below for the pipeline to run.
```suggestion
                - task: 1ES.PublishNuget@1
```

File: .azure-pipelines/ci-build.yml [hunk 3]
```diff
@@ -21,248 +18,190 @@ variables:
...
+                - task: DownloadPipelineArtifact@2
+                  displayName: Download nupkg from artifacts
+                  inputs:
+                    artifact: Nugets
+                    source: current
+                - task: PowerShell@2
+                  displayName: 'Extract release information to pipeline'
+                  inputs:
+                    targetType: 'filePath'
+                    filePath: $(Pipeline.Workspace)\Nugets\scripts\GetNugetPackageVersion.ps1
```
Comment (andrueastman) [inline .azure-pipelines/ci-build.yml:199]:
```suggestion
                    filePath: $(Pipeline.Workspace)\scripts\GetNugetPackageVersion.ps1
```
I believe artifacts will be extracted at the root.

File: .azure-pipelines/ci-build.yml [hunk 4]
```diff
@@ -21,248 +18,190 @@ variables:
...
+                    pwsh: true
+                    arguments: '-packageDirPath "$(Pipeline.Workspace)/Nugets/"'
+                - task: 1ES.PublishNuget@1
+                  displayName: 'NuGet push'
+                  inputs:
+                    packagesToPush: '$(Pipeline.Workspace)/Nugets/Microsoft.Kiota.Cli.Commons.*.nupkg'
+                    nuGetFeedType: external
+                    publishFeedCredentials: 'Kiota Nuget Connection'
```
Comment (andrueastman) [inline .azure-pipelines/ci-build.yml:207]:
```suggestion
                    packagesToPush: '$(Pipeline.Workspace)/Microsoft.Kiota.Cli.Commons.*.nupkg'
                    nuGetFeedType: external
                    publishFeedCredentials: 'Kiota Nuget Connection'
                    packageParentPath: '$(Pipeline.Workspace)'
```

File: .azure-pipelines/ci-build.yml [hunk 5]
```diff
@@ -21,248 +18,190 @@ variables:
...
-          BinSkim: true
-          CredScan: true
-          PoliCheck: true
-
-      - task: UseDotNet@2
-        displayName: 'Add .NET 2'
-        inputs:
-          version: 2.x
-
-      - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@1
```
Comment (baywet) [inline .azure-pipelines/ci-build.yml:108]: the code sign task should remain

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**